mro/table.py

```diff
@@ -98,14 +102,15 @@ def _convert_numpy_types_to_python(values):
         return values
 
     @classmethod
+    @retry(wait=wait_random_exponential(), stop=stop_after_attempt(MAX_ATTEMPTS),
+           reraise=True, before_sleep=before_sleep_log(logger, logging.WARNING))
     def select(cls, clause=None, *format_args):
         format_args = list(format_args)
-        cursor = cls._get_cursor()
         if clause is None:
             sql = "select * from \"{}\";".format(cls.__name__)
         else:
             sql = "select * from \"{}\" where {};".format(cls.__name__, clause)
```

Comment:
In all of these functions, there was the pattern of creating a cursor and then passing it to `_execute_sql` to use, rather than just letting `_execute_sql` create one. The benefit of the latter option is that the cursor creation happens within the scope of the `psycopg2` lock. Therefore, when a disconnect occurs, only a single thread will trigger the reconnect - whichever one happens to have the lock at that time. 

Without this change, you end up with potentially many threads all having cursors that are no longer valid (since the underlying connection was closed) => they all try to use the cursors and fail => they all trigger retries => they all try to recreate the cursor => they all trigger reconnects at the same time. This doesn't work at all given `mro`'s use of a global connection object.